core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -0,0 +1,120 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for a set of audio controls for a specific
+ * audio translation in the learner view.
+ */
+
+oppia.directive('audioControls', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        audioTranslations: '='
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/exploration_player/' +
+        'audio_controls_directive.html'),
+      controller: [
+        '$scope', 'AudioTranslationManagerService', 'AudioPlayerService',
+        function(
+            $scope, AudioTranslationManagerService, AudioPlayerService) {
+          var filenameOfLastStartedAudio;
+
+          $scope.AUDIO_SETTINGS_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/settings.svg'));
+
+          var PLAY_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/speaker-not-playing.svg'));
+
+          var PAUSE_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/speaker-playing.svg'));
+
+          $scope.REWIND_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/rewind-five.svg'));
+
+          $scope.playPauseAudioTranslation = function() {
+            // TODO(tjiang11): Change from on-demand loading to pre-loading.
+
+            // TODO(tjiang11): On first play, ask learner to pick language
+            // and subsequently for confirmation to use bandwidth 
+            // to download audio files.
+
+            $scope.extraAudioControlsAreShown = true;
+
+            if (!AudioPlayerService.isPlaying()) {
+              if (AudioPlayerService.isTrackLoaded() &&
+                  isRequestForSameAudioAsLastTime()) {
+                AudioPlayerService.play();
+              } else {
+                loadAndPlayAudioTranslation();
+              }
+            } else {
+              AudioPlayerService.pause();
+              if (!isRequestForSameAudioAsLastTime()) {
```

Comment:
This should only activate when a user has clicked a "play" button on an audio translation that is different from the current audio being played. This way, the audio player pauses the existing audio and immediately starts playing the newly requested audio translation.